core/src/test/scala/org/apache/spark/status/AppStatusListenerSuite.scala

```diff
@@ -1190,6 +1190,65 @@ class AppStatusListenerSuite extends SparkFunSuite with BeforeAndAfter {
     assert(appStore.asOption(appStore.lastStageAttempt(3)) === None)
   }
 
+  test("stage and job metrics should be updated when stage completion event" +
```

Comment:
"update metrics for tasks that finish late". Same thing, much shorter. Could also add the bug number to the test name.